lib/components/segmented-control/base.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { RefObject } from "react";
+import clsx from "clsx";
+import { TRegularSizes } from "../../types";
+import { Segment } from "./segment";
+import "./segmented-control.scss";
+
+export interface SegmentedControlProps {
+    className?: string;
+    options: Array<{
+        disabled?: boolean;
+        icon?: string | React.ReactNode;
+        label?: string;
+        selected?: boolean;
+    }>;
+    onChange?: (selectedItemIndex: number) => void;
+    hasContainerWidth?: boolean;
+    size?: TRegularSizes;
+}
+
+export const SegmentedControl = ({
+    className,
+    options = [],
+    onChange,
+    hasContainerWidth,
+    size = "md",
+}: SegmentedControlProps) => {
+    const handleKeyboardEvents = (
+        e: React.KeyboardEvent<HTMLButtonElement>,
+        idx: number,
+        ref: RefObject<HTMLButtonElement>,
+    ) => {
+        if (!options[idx].disabled && (e.key === "Enter" || e.key === " ")) {
```

Comment:
what if the button in segment will be disabled?

Reply:
it won't be selected :)